### PR TITLE
[#373] Fix make manifest on MacOS.

### PR DESCRIPTION
**Pull request by davidlanouette**
Mac and linux have different versions of sed (bsd and gnu respectively). The '-i' flag works slightly differently - just enough so they can't be used interchangeably.

This MR switches to a simple perl command to accomplish the same function.

### CODE REVIEW

**Comment by davidlanouette** [comment]
This has been tested on both MacOS and Fedora.

**Comment by davidlanouette** [comment]
NOTE: While this fixes the reported problem, I'm wondering why the 'creationTime' field needs to be removed at all? Was this causing some issue previously?

**Comment by gaohoward** [comment]
maybe use python instead of perl ?

**Comment by brusdev** [comment]
@davidlanouette the creationTime fields were removed because they were not required and they were stale. I would not add other env requirements for this script neither python or perl.

**Comment by davidlanouette** [comment]
> the creationTime fields were removed because they were not required and they were stale.

I asked because the current version of the `controller-gen` tool (v0.11.0) will add `creationTimestamp: null` to the metadata.  I'm not sure what v0.7.0 will do.

**Comment by davidlanouette** [comment]
If you think that field still needs to be removed, I'll revert to using `sed` again.  But, I'll have to add a conditional to check which version of `sed` is installed, and configure the command line based on that.

Do you object to pulling that logic out into a full script in the `hack` directory?

File: Makefile [hunk 1]
```diff
@@ -101,13 +101,13 @@ ifeq ($(ENABLE_WEBHOOKS),true)
 ## v2alpha3, v2alpha4 and v2alpha3 requires allowDangerousTypes=true because they use float32 type
 	cd config/manager && $(KUSTOMIZE) edit add resource webhook_secret.yaml 
 	$(CONTROLLER_GEN) rbac:roleName=$(OPERATOR_CLUSTER_ROLE_NAME) crd:allowDangerousTypes=true webhook paths="./..." output:crd:artifacts:config=config/crd/bases
-	find config -type f -exec sed -i -e '/creationTimestamp/d' {} \;
+	find config -type f -exec ./hack/inplace_sed.sh -e '/creationTimestamp/d' {} \;
```
Comment (brusdev) [inline Makefile:104]: The `-i.bak` seems working on both `BSD` and `GNU` systems:
```suggestion
	find config -type f -exec sed -i.bak -e '/creationTimestamp/d' {} \; -exec rm {}.bak \;
```

File: Makefile [hunk 2]
```diff
@@ -213,11 +213,11 @@ bundle: manifests operator-sdk kustomize ## Generate bundle manifests and metada
 	$(OPERATOR_SDK) generate kustomize manifests -q --package $(BUNDLE_PACKAGE)
 	cd config/manager && $(KUSTOMIZE) edit set image controller=$(IMG)
 	$(KUSTOMIZE) build config/manifests | $(OPERATOR_SDK) generate bundle -q --overwrite --package $(BUNDLE_PACKAGE) --version $(VERSION) $(BUNDLE_METADATA_OPTS)
-	sed -i '/creationTimestamp/d' ./bundle/manifests/*.yaml
+	./hack/inplace_sed.sh '/creationTimestamp/d' ./bundle/manifests/*.yaml
```
Comment (brusdev) [inline Makefile:216]: The -i.bak seems working on both BSD and GNU systems:
```suggestion
	sed -i.bak '/creationTimestamp/d' ./bundle/manifests/*.yaml && rm ./bundle/manifests/*.bak
```

**Comment by davidlanouette** [comment]
@brusdev Thanks for the suggestion.

**Comment by brusdev** [comment]
@davidlanouette thanks for commitment, I'll merge after fixing this last failure caused by `go install`

File: Makefile [hunk 3]
```diff
@@ -203,7 +203,7 @@ TMP_DIR=$$(mktemp -d) ;\
 cd $$TMP_DIR ;\
 go mod init tmp ;\
 echo "Downloading $(2)" ;\
-GOBIN=$(PROJECT_DIR)/bin go get $(2) ;\
+GOBIN=$(PROJECT_DIR)/bin go install $(2) ;\
```
[the remark this answers is not in the document]
Comment (davidlanouette) [reply Makefile:206]: Sorry, I didn't mean to commit that change.  (It is required for newer versions of Go, but I know that's being worked on in a different PR.)

I'll remove it now.

[the remark this answers is not in the document]
Comment (davidlanouette) [reply Makefile]: Done.

**Comment by brusdev** [comment]
@davidlanouette thanks for your commitment and your patience

**Pull request merged**